src/gui/guiiconprovider.cpp

```diff
@@ -71,13 +72,20 @@ QIcon GuiIconProvider::getIcon(const QString &iconId, const QString &fallback) c
         QIcon icon = QIcon::fromTheme(iconId);
         if (icon.name() != iconId)
             icon = QIcon::fromTheme(fallback, QIcon(IconProvider::getIconPath(iconId)));
-        icon = generateDifferentSizes(icon);
         return icon;
     }
 #else
     Q_UNUSED(fallback)
 #endif
-    return QIcon(IconProvider::getIconPath(iconId));
+    // cache to avoid rescaling svg icons
+    static QHash<QString, QIcon> iconCache;
```

Comment:
Maybe we better add an index.theme file and use `QIcon::fromTheme()`, that caches icons?

Reply:
I'm not sure, I think I'll leave it as-is for you to hack on later.

Reply:
http://code.qt.io/cgit/qt/qtbase.git/tree/src/gui/image/qicon.cpp#n1254

Maybe it would make sense to push the transition to `QIcon::fromTheme()` first then? I could try to make a PR today.

Reply:
@Chocobo1: take a look, please: #9295

Reply:
>@Chocobo1: take a look, please: #9295

I've looked a bit, however I find the level of changes are out of my (or contributor) jurisdiction.
I somehow would prefer merging this PR first, it is small and (IMO) ready and also I'm not too fond of rebasing this PR over large changes.